crates/core/tedge_agent/src/agent.rs

```diff
@@ -225,10 +225,24 @@ impl SmAgent {
         })
     }
 
+    pub fn get_log_dir(&self) -> PathBuf {
+        self.config.log_dir.clone()
+    }
+
     #[instrument(skip(self), name = "sm-agent")]
-    pub async fn init(&mut self, config_dir: PathBuf) -> Result<(), anyhow::Error> {
+    pub async fn init(
+        &mut self,
+        config_dir: PathBuf,
+        log_path: PathBuf,
+    ) -> Result<(), anyhow::Error> {
+        // `config_dir` by default is `/etc/tedge` (or whatever the user sets with --config-dir)
+        // `log_path` is by default `/var/log` (or whatever the user sets with tedge config set logs.path)
+        let cfg_dir = config_dir.as_path().display().to_string();
+        let log_path = log_path.as_path().display().to_string();
```

Comment:
Is this necessary? `PathBuf` can be made `Display`, so the interpolation should work without the full conversion to string!

```suggestion
        let cfg_dir = config_dir.display();
        let log_path = log_path.display();
```

You might want to rename the variables though.